### PR TITLE
feat: improve performance of tlsPeerIdentifier to O(1)

### LINKED ISSUE #12709: feat: improve performance of tlsPeerIdentifier to O(1)

### PR DESCRIPTION
**Pull request by kfa-aguda**
closes #12709. 

This PR improves the certificate lookup performance of a newly connected network peer.

### CODE REVIEW

**Comment by lpetrovic05** [comment]
@kfa-aguda @cody-littley 
This approach makes dynamic changes to the peer list more difficult. We should probably discuss an alternative approach

**Comment by cody-littley** [comment]
@lpetrovic05 

> This approach makes dynamic changes to the peer list more difficult. We should probably discuss an alternative approach

When we want things to be dynamic, I'm guessing we are going to need a method that says "hey, use this new peer list".

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkPeerIdentifier.java [hunk 1]
```diff
@@ -46,52 +45,49 @@ public class NetworkPeerIdentifier {
      */
     private final RateLimitedLogger noPeerFoundLogger;
 
-    private final RateLimitedLogger multiplePeersFoundLogger;
+    // a mapping of X500Principal and their peers
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkPeerIdentifier.java:48]: Member docs should use /* */ comment style (only need be addressed in this PR if there are other needed changes)

Comment (kfa-aguda) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkPeerIdentifier.java:48]: thank you. Will take care of it in a subsequent PR.

**Comment by poulok** [comment]
> When we want things to be dynamic, I'm guessing we are going to need a method that says "hey, use this new peer list".

Why not just create the peer info as a Map in the first place and pass it into the method instead of the constructor?

**Review by edward-swirldslabs — APPROVED** [review]

**Comment by kfa-aguda** [comment]
> > When we want things to be dynamic, I'm guessing we are going to need a method that says "hey, use this new peer list".
> 
> Why not just create the peer info as a Map in the first place and pass it into the method instead of the constructor?

We discussed that actually but as it's generally not the best to pass around mutable collections, I'd have to write some wrapper around the map with a get. Whilst that also works, this is simpler.

**Pull request merged**